src/pages/TeacherDashboard/TeacherDashboard.jsx

```diff
@@ -1,63 +1,92 @@
 import React, {useState, useEffect} from 'react';
 import styles from "./TeacherDashboardStyles.module.css"
 import SideBar from "../../components/SideBar/SideBar";
+import {useNavigate} from "react-router-dom";
 
 const TeacherDashboard = () => {
     const apiUrl = import.meta.env.VITE_API_BASE_URL;
     const [courses, setCourses] = useState([]);
+    const token = localStorage.getItem('token');
+    const navigate = useNavigate();
 
     useEffect(() => {
-        fetch(`${apiUrl}/course`,
+        if (!token) {
+            navigate('/');
+            return
+        }
+
+        fetch(`${apiUrl}/course/teacher_courses`,
             {
                 method: 'GET',
                 headers: {
                     'Content-Type': 'application/json',
                     'Accept': 'application/json',
-                    'Authorization': `Bearer ${localStorage.getItem('token')}`
+                    'Authorization': `Bearer ${token}`
                 }
             })
             .then(response => {
-            return response.json();
-        }).then(data => {
-            setCourses(data);
-        }).catch(error => {
-            console.log(error);
-        });
+                if (!response.ok) {
+                    throw new Error('Network response was not ok');
+                }
+                return response.json();
+            }).then(data => {
+                setCourses(data);
+            }).catch(error => {
+                console.log(error);
+                navigate('/');
+            });
     }, []);
 
+    const totalPoints = (lessons) => {
+        return lessons.reduce((pointSum, {points}) => pointSum + points, 0);
+    };
+
+    if (!token) return null;
```

Comment:
We should not just return null here, which will make our page go blank. In theory, the useEffect should navigate us back if the token is not valid, but we should still render a valid page until that happens. The courses array should be empty so your layout might work anyways and just not show any course content, which is fine if so.